src/main/java/com/googlecode/objectify/impl/translate/SerializeTranslatorFactory.java

```diff
@@ -75,14 +131,63 @@ protected Blob saveValue(Object value, boolean index, SaveContext ctx, Path path
 					oos.writeObject(value);
 					oos.close();
 
-					return new Blob(baos.toByteArray());
+					byte[] bytes = baos.toByteArray();
+
+					if (serializeAnno.encrypt()) {
+						//attempt to encrypt
+						bytes = encrypt(baos.toByteArray());
+					}
+
+					return new Blob(bytes);
 
 				} catch (IOException ex) {
 					path.throwIllegalState("Unable to serialize " + value, ex);
 					return null;	// never gets here
 				}
 			}
 
+
+			/**
+			 * encrypt value
+			 */
+			private byte[] encrypt (byte[] input)  {
+				SortedMap<String, SecretKeySpec> keyMap = this.getKeyMap();
+
+				if (keyMap == null) {
```

Comment:
missing empty map check, otherwise you'll get a fancy NPE when doing `keyMap.firstKey()`. Same in decrypt.